src/Elastic.Markdown/BuildContext.cs

```diff
@@ -34,6 +34,9 @@ public record BuildContext
 	// This property is used to determine if the site should be indexed by search engines
 	public bool AllowIndexing { get; init; }
 
+	// This property is used for the canonical URL
+	public string? CanonicalBaseUrl { get; init; }
```

Comment:
`Uri` instead of string

Reply:
https://github.com/elastic/docs-builder/pull/776/commits/bb378c177a577aba1b2a052a43cb79f7156d7b6f